pkgs/dkg/drand.go

```diff
@@ -214,77 +360,199 @@ func (o *LocalOwner) PostDKG(res *dkg.OptionResult) error {
+	}
+
+	encodedOutput, err := out.Encode()
 	if err != nil {
 		o.broadcastError(err)
 		return err
 	}
-	if err = shareSecretDecrypted.SetHexString(string(decryptedSharePrivateKey)); err != nil {
+
+	tsMsg := &wire.Transport{
+		Type:       wire.OutputMessageType,
+		Identifier: o.Data.ReqID,
+		Data:       encodedOutput,
+	}
+
+	o.Broadcast(tsMsg)
+	close(o.Done)
+	return nil
+}
+
+func (o *LocalOwner) PostReshare(res *dkg.OptionResult) error {
+	if res.Error != nil {
+		o.Logger.Error("DKG ceremony returned error: ", zap.Error(res.Error))
+		o.broadcastError(res.Error)
+		return res.Error
+	}
+	o.Logger.Info("DKG resharing ceremony finished successfully")
+	// Store result share a instance
+	o.SecretShare = res.Result.Key
+	// encode priv share
+	secret := &DistKeyShare{}
+	var commits []byte
+	for _, point := range o.SecretShare.Commits {
+		b, _ := point.MarshalBinary()
+		commits = append(commits, b...)
+	}
+	secret.Commits = commits
+	secterPoint, err := o.SecretShare.Share.V.MarshalBinary()
+	if err != nil {
 		o.broadcastError(err)
 		return err
 	}
-
-	if !bytes.Equal(shareSecretDecrypted.Serialize(), secretKeyBLS.Serialize()) {
+	secret.Share.V = secterPoint
+	secret.Share.I = o.SecretShare.Share.I
+	bin, err := secret.Encode()
+	if err != nil {
 		o.broadcastError(err)
 		return err
 	}
+	encBin, err := o.EncryptSecretDB(bin)
+	if err != nil {
+		o.broadcastError(err)
+		return err
+	}
+	err = o.DB.Set([]byte("secret"), o.Data.ReqID[:], encBin)
+	if err != nil {
+		o.broadcastError(err)
+		return err
+	}
+	// Get validator BLS public key from result
+	validatorPubKey, err := crypto.ResultToValidatorPK(res.Result, o.Suite.G1().(dkg.Suite))
+	if err != nil {
+		o.broadcastError(err)
+		return err
+	}
+	o.Logger.Debug("Validator`s public key %x", zap.String("key", fmt.Sprintf("%x", validatorPubKey.Serialize())))
 
-	o.Logger.Debug("Encrypted share", zap.String("share", fmt.Sprintf("%x", ciphertext)))
-	o.Logger.Debug("Withdrawal Credentials", zap.String("creds", fmt.Sprintf("%x", o.data.init.WithdrawalCredentials)))
-	o.Logger.Debug("Fork Version", zap.String("v", fmt.Sprintf("%x", o.data.init.Fork[:])))
-	o.Logger.Debug("Domain", zap.String("bytes", fmt.Sprintf("%x", ssvspec_types.DomainDeposit[:])))
-
-	// Sign root
-	depositRootSig, signRoot, err := crypto.SignDepositData(secretKeyBLS, o.data.init.WithdrawalCredentials[:], validatorPubKey, GetNetworkByFork(o.data.init.Fork), MaxEffectiveBalanceInGwei)
-	o.Logger.Debug("Root", zap.String("", fmt.Sprintf("%x", signRoot)))
-	// Validate partial signature
-	val := depositRootSig.VerifyByte(secretKeyBLS.GetPublicKey(), signRoot)
-	if !val {
+	// Get BLS partial secret key share from DKG
+	secretKeyBLS, err := crypto.ResultToShareSecretKey(res.Result)
+	if err != nil {
 		o.broadcastError(err)
-		return fmt.Errorf("partial deposit root signature is not valid %x", depositRootSig.Serialize())
+		return err
+	}
+	// Store secret if requested
+	if StoreShare {
```

Comment:
same comment as above

Reply:
Moved higher level to operator/state package.